java/java.source.base/src/org/netbeans/modules/java/source/parsing/JavacParser.java

```diff
@@ -1078,6 +1086,23 @@ private static JavacTaskImpl createJavacTask(
         return task;
     }
 
+    private static String useRelease(final String requestedSource, com.sun.tools.javac.code.Source validatedSourceLevel) {
+        if (requestedSource == null || validatedSourceLevel == null) {
+            return null;
+        }
+        com.sun.tools.javac.code.Source sourceLevel = com.sun.tools.javac.code.Source.lookup(requestedSource);
+        if (sourceLevel == null) {
```

Comment:
Working with OpenJDK sources out of the box is an important use-case for NetBeans. By creating the `useRelease` method and adding this `sourceLevel == null` check I am able to open `java.base` from https://github.com/openjdk/jdk/commit/c2c0cb2a4372d78658326461562363de9a1a194f without major errors.

Reply:
These are the non-fatal errors I see on console:

```
java.lang.NullPointerException
org.netbeans.modules.java.editor.base.semantic.UnusedDetector.isUnusedInPkg(UnusedDetector.java:322)
org.netbeans.modules.java.editor.base.semantic.UnusedDetector.findUnused(UnusedDetector.java:138)
org.netbeans.modules.java.editor.base.semantic.SemanticHighlighterBase.process(SemanticHighlighterBase.java:185)
```
```
java.lang.NullPointerException
org.netbeans.modules.java.source.indexing.VanillaCompileWorker$3.visitVariable(VanillaCompileWorker.java:585)
org.netbeans.modules.java.source.indexing.VanillaCompileWorker$3.visitVariable(VanillaCompileWorker.java:535)
com.sun.tools.javac.tree.JCTree$JCVariableDecl.accept(JCTree.java:1045)
com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:86)
org.netbeans.modules.java.source.indexing.VanillaCompileWorker$3.scan(VanillaCompileWorker.java:886)
```

```
WARNING [org.netbeans.modules.java.source.usages.BinaryAnalyser]: Invalid class file format: file:/home/devel/NetBeansProjects/netbeans/nbbuild/netbeans/java/modules/ext/fakeJdkClasses.zip!/java/lang/AutoCloseable.class
INFO [org.netbeans.modules.java.source.usages.BinaryAnalyser]: Class File Exception Details
java.io.EOFException
        at java.base/java.io.DataInputStream.readInt(DataInputStream.java:397)
        at org.netbeans.modules.classfile.ClassFile.loadClassHeader(ClassFile.java:183)
        at org.netbeans.modules.classfile.ClassFile.load(ClassFile.java:168)
Caused: org.netbeans.modules.classfile.InvalidClassFormatException
        at org.netbeans.modules.classfile.ClassFile.load(ClassFile.java:174)
        at org.netbeans.modules.classfile.ClassFile.<init>(ClassFile.java:126)
        at org.netbeans.modules.classfile.ClassFile.<init>(ClassFile.java:73)
        at org.netbeans.modules.java.source.usages.BinaryAnalyser.analyse(BinaryAnalyser.java:670)
```